src/index.ts

```diff
@@ -41,7 +41,8 @@ process.on("SIGINT", async (_: string, code: number) => {
                     } else {
                         process.stderr.write(chalk`{red ${e.stack ?? e}}\n`);
                     }
-                    cleanupJobResources(jobs).finally(process.exit(1));
```

Comment:
with this,

process seemed to exit before cleanup, any idea why ? 


Run this, 
```yaml
cache:
  key:
    files:
      - /foo
  paths:
    - ${CI_PROJECT_DIR}/node_modules
job0:
  image: bash
  script: echo hi
```

and then `docker ps -a`

Reply:
Using the following .gitlab-ci.yml
```yaml
cache:
  key:
    files:
      - /foo
  paths:
    - ${CI_PROJECT_DIR}/node_modules
job0:
  image: bash
  script: echo hi
 ```
 
 and running `gitlab-ci-local`
 
 the cleanup works for the `await` syntax but not `finally` syntax

Reply:
await is a blocking statement

cleanupJobResources(jobs).finally(process.exit(1)); doesn't block anything

Reply:
I imagine the program shuts down, before even getting to the callback function in finally.
So it kindda makes sense.